apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.js

```diff
@@ -81,256 +95,183 @@ router.post(
-            /^git@github.com:\/?(.+?)(\.git)?\/?$/,
-          );
-          if (GHfound) {
-            res.locals.questionGHLink =
-              'https://github.com/' +
-              GHfound[1] +
-              '/tree/master/questions/' +
-              res.locals.question.qid;
-          }
-        } else if (res.locals.course.example_course) {
-          res.locals.questionGHLink = `https://github.com/PrairieLearn/PrairieLearn/tree/master/exampleCourse/questions/${res.locals.question.qid}`;
-        }
-        callback(null);
-      },
-      (callback) => {
-        sqldb.queryOneRow(sql.qids, { course_id: res.locals.course.id }, (err, result) => {
-          if (ERR(err, callback)) return;
-          res.locals.qids = result.rows[0].qids;
-          callback(null);
-        });
-      },
-      (callback) => {
-        sqldb.query(
-          sql.select_assessments_with_question_for_display,
-          { question_id: res.locals.question.id },
-          (err, result) => {
-            if (ERR(err, callback)) return;
-            res.locals.a_with_q_for_all_ci = result.rows[0].assessments_from_question_id;
-            callback(null);
-          },
-        );
-      },
-      async () => {
-        res.locals.sharing_enabled = await features.enabledFromLocals(
-          'question-sharing',
-          res.locals,
-        );
+    res.locals.questionGHLink = null;
+    if (res.locals.course.repository) {
+      const GHfound = res.locals.course.repository.match(/^git@github.com:\/?(.+?)(\.git)?\/?$/);
+      if (GHfound) {
+        res.locals.questionGHLink =
+          'https://github.com/' + GHfound[1] + '/tree/master/questions/' + res.locals.question.qid;
+      }
+    } else if (res.locals.course.example_course) {
+      res.locals.questionGHLink = `https://github.com/PrairieLearn/PrairieLearn/tree/master/exampleCourse/questions/${res.locals.question.qid}`;
+    }
 
-        if (res.locals.sharing_enabled) {
-          let result = await sqldb.queryAsync(sql.select_sharing_sets, {
-            question_id: res.locals.question.id,
-            course_id: res.locals.course.id,
-          });
-          res.locals.sharing_sets_in = result.rows.filter((row) => row.in_set);
-          res.locals.sharing_sets_other = result.rows.filter((row) => !row.in_set);
-        }
-      },
-      async () => {
-        res.locals.editable_courses = await selectCoursesWithEditAccess({
-          user_id: res.locals.user.user_id,
-          is_administrator: res.locals.is_administrator,
-        });
-      },
-    ],
-    (err) => {
-      if (ERR(err, next)) return;
-      res.locals.infoPath = encodePath(
-        path.join('questions', res.locals.question.qid, 'info.json'),
-      );
-      res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-    },
-  );
-});
+    res.locals.qids = await sqldb.queryRow(
+      sql.qids,
+      { course_id: res.locals.course.id },
+      z.array(z.string().nullable()),
```

Comment:
As with `SelectedAssessmentsSchema`, you may be able to change the query + schema to return a normal array of multiple rows.

Reply:
Removed the array from query and updated to `queryRows` in [15e6a55](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/15e6a551ace1b217ed76fec735c5306f682080d1).